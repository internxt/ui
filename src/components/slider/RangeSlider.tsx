import { ChangeEvent } from 'react';
import '../../styles/RangeSlider.css';

interface RangeSliderProps {
  value: number;
  min?: number;
  max: number;
  step?: number;
  className?: string;
  color?: string;
  onChange: (value: number) => void;
  disabled?: boolean;
  ariaLabel?: string;
}

export const RangeSlider = ({
  value,
  min = 0,
  max,
  step = 1,
  className,
  onChange,
  disabled = false,
  ariaLabel = 'Range slider',
}: RangeSliderProps) => {
  const percentage = (value / max) * 100;
  const sliderBackground = `linear-gradient(to right, #3264fe ${percentage}%, #d5d5d5 ${percentage}%)`;

  const handleChange = (newValue: number) => {
    onChange(newValue);
  };

  return (
    <div className={className}>
      <input
        id="my-slider"
        type="range"
        min={min}
        max={max}
        value={value}
        step={step}
        onInput={(e: ChangeEvent<HTMLInputElement>) => handleChange(Number(e.target.value))}
        disabled={disabled}
        aria-label={ariaLabel}
        style={{ background: sliderBackground }}
      />
    </div>
  );
};
