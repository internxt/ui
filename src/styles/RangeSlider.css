.custom-range-slider {
  width: 100%;
  height: 6px;
  background-color: #ffffff;
  border-radius: 50%;
  outline: none;
  position: relative;
}

input[type="range"] {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: '100%';
  background: #d5d5d5;
  border-radius: 16px;
  outline: none;
}

input[type="range"]::-webkit-slider-runnable-track {
  -webkit-appearance: none;
  height: 6px;
}

input[type="range"]::-moz-track {
  appearance: none;
  height: 6px;

}

input[type="range"]::-ms-track {
  appearance: none;
  height: 6px;
}

input[type="range"]::-webkit-slider-thumb {
  -webkit-appearance: none;
  height: 24px;
  width: 24px;
  background-color: #3264fe;
  border-radius: 50%;
  pointer-events: auto;
  cursor: pointer;
  margin-top: -9px;
  border: none;
}

input[type="range"]::-moz-range-thumb {
  appearance: none;
  height: 24px;
  width: 24px;
  background-color: #3264fe;
  border-radius: 50%;
  pointer-events: auto;
  cursor: pointer;
  margin-top: -9px;
  border: none;
}

input[type="range"]::-ms-thumb {
  appearance: none;
  height: 24px;
  width: 24px;
  background-color: #3264fe;
  border-radius: 50%;
  pointer-events: auto;
  cursor: pointer;
  margin-top: -9px;
  border: none;
}