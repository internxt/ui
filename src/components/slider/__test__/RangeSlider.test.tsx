import React from 'react';
import { describe, it, expect, vi } from 'vitest';
import { render, screen, fireEvent } from '@testing-library/react';
import { RangeSlider } from '../RangeSlider';

describe('RangeSlider component', () => {
  it('RangeSlider onChange should work correctly', () => {
    const handleChange = vi.fn();
    render(<RangeSlider value={50} max={100} onChange={handleChange} />);

    const slider = screen.getByRole('slider');

    // Cambiamos el valor del slider
    fireEvent.change(slider, { target: { value: '75' } });

    expect(handleChange).toHaveBeenCalledOnce();
    expect(handleChange).toHaveBeenCalledWith(75);
  });

  it('RangeSlider component should set the progress correctly on mount', () => {
    render(<RangeSlider value={50} max={100} onChange={() => {}} />);
    const slider = screen.getByRole('slider');

    // Comprobamos que la propiedad CSS --progress se haya establecido correctamente
    expect(slider.style.getPropertyValue('--progress')).toBe('50%');
  });

  it('RangeSlider component should render correctly', () => {
    const { container } = render(<RangeSlider value={50} max={100} onChange={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
