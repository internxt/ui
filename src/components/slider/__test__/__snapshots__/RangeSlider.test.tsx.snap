// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RangeSlider component > RangeSlider component should render correctly 1`] = `
<div>
  <div>
    <input
      aria-label="Range slider"
      id="my-slider"
      max="100"
      min="0"
      style="background: linear-gradient(to right, #3264fe 50%, #d5d5d5 50%);"
      type="range"
      value="50"
    />
  </div>
</div>
`;
